### ISSUE TITLE
Extend the `Chain` trait to enable listening for IBC events from non-Cosmos SDK chains, eg. Substrate

### ISSUE THREAD

**Issue by DaviRain-Su** [informalsystems/hermes#1456]
## Crate

None

## Summary

init_event_monitor Support multiply chain to event mointor

https://github.com/informalsystems/ibc-rs/blob/8944f94815c35b2d3297971aa437900d259dc75a/relayer/src/chain.rs#L99
```
/// Initializes and returns the event monitor (if any) associated with this chain.
    fn init_event_monitor(
        &self,
        rt: Arc<TokioRuntime>,
    ) -> Result<(EventReceiver, TxMonitorCmd), Error>;
```

https://github.com/informalsystems/ibc-rs/blob/8944f94815c35b2d3297971aa437900d259dc75a/relayer/src/event/monitor.rs#L132
```
pub struct EventMonitor {
    chain_id: ChainId,
    /// WebSocket to collect events from
    client: WebSocketClient,
    /// Async task handle for the WebSocket client's driver
    driver_handle: JoinHandle<()>,
    /// Channel to handler where the monitor for this chain sends the events
    tx_batch: channel::Sender<Result<EventBatch>>,
    /// Channel where to receive client driver errors
    rx_err: mpsc::UnboundedReceiver<tendermint_rpc::Error>,
    /// Channel where to send client driver errors
    tx_err: mpsc::UnboundedSender<tendermint_rpc::Error>,
    /// Channel where to receive commands
    rx_cmd: channel::Receiver<MonitorCmd>,
    /// Node Address
    node_addr: Url,
    /// Queries
    event_queries: Vec<Query>,
    /// All subscriptions combined in a single stream
    subscriptions: Box<SubscriptionStream>,
    /// Tokio runtime
    rt: Arc<TokioRuntime>,
}

```

## Problem Definition

Recently I am doing support for substrate relayer support, for event listening here, it seems that there is no support for multiple chains like ChainEndpoint defines an interface, here for EventMonitor to listen to events suitable for `tendermint-rpc` strong binding, no implementation and different chains between the loose coupling. Is there a good implementation to achieve this part of the function, because I found that this EventMointor in the implementation of `start` and `listen` command is related.

## Proposal

None

## Acceptance Criteria

None

____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [x] Appropriate milestone (priority) applied
- [x] Appropriate contributors tagged
- [x] Contributor assigned/self-assigned

**Comment by romac** [comment]
@DaviRain-Su What do you think of the refactoring in #2865? Does this help with your use case?

**Comment by DaviRain-Su** [comment]
It looks good, but there are still some components that are strongly related to tendermint that I wish could be abstracted

**Comment by romac** [comment]
Closed by #2865: https://github.com/informalsystems/hermes/pull/2865#discussion_r1036039050
